### ISSUE TITLE
clang-format v18.1.2 crash when set `ColumnLimit: 0`

### ISSUE THREAD

**Issue by valord577** [llvm/llvm-project#86281]
```text
[user@host ~]% clangd --version
clangd version 18.1.2 (http://git.linaro.org/toolchain/jenkins-scripts.git e5def089cd9f5aa71524f82fef301ca66eaa38d2)
Features: linux
Platform: aarch64-unknown-linux-gnu
```

> vscode output

```text
I[21:06:14.791] clangd version 18.1.2 (http://git.linaro.org/toolchain/jenkins-scripts.git e5def089cd9f5aa71524f82fef301ca66eaa38d2)
I[21:06:14.791] Features: linux
I[21:06:14.791] PID: 27757
I[21:06:14.791] Working directory: /root/code/ffmpeg-sv
I[21:06:14.791] argv[0]: /usr/local/bin/clangd
I[21:06:14.791] argv[1]: --compile_args_from=filesystem
I[21:06:14.791] argv[2]: --background-index
I[21:06:14.791] argv[3]: --clang-tidy
I[21:06:14.791] argv[4]: --completion-parse=always
I[21:06:14.791] argv[5]: --completion-style=detailed
I[21:06:14.791] argv[6]: --function-arg-placeholders=0
I[21:06:14.791] argv[7]: --header-insertion=iwyu
I[21:06:14.791] argv[8]: --import-insertions
I[21:06:14.791] argv[9]: --limit-references=0
I[21:06:14.791] argv[10]: --limit-results=0
I[21:06:14.791] argv[11]: --rename-file-limit=0
I[21:06:14.791] argv[12]: --pch-storage=memory
I[21:06:14.791] argv[13]: --offset-encoding=utf-8
I[21:06:14.791] Starting LSP over stdin/stdout
I[21:06:14.792] <-- initialize(0)
I[21:06:14.795] --> reply:initialize(0) 2 ms
I[21:06:14.801] <-- initialized

.... [LSP logs]

I[21:08:25.809] <-- textDocument/formatting(11)
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
 #0 0x0000005574c8697c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/usr/local/bin/clangd+0x5e697c)
 #1 0x0000005574c84b0c llvm::sys::RunSignalHandlers() (/usr/local/bin/clangd+0x5e4b0c)
 #2 0x0000005574c87408 SignalHandler(int) Signals.cpp:0:0
 #3 0x0000007f8cf987d0 (linux-vdso.so.1+0x7d0)
 #4 0x000000557665a598 clang::format::WhitespaceManager::getCells(unsigned int, unsigned int) (/usr/local/bin/clangd+0x1fba598)
 #5 0x0000005576659b44 clang::format::WhitespaceManager::alignArrayInitializers(unsigned int, unsigned int) (/usr/local/bin/clangd+0x1fb9b44)
 #6 0x00000055766568d4 clang::format::WhitespaceManager::generateReplacements() (/usr/local/bin/clangd+0x1fb68d4)
 #7 0x00000055766057b0 clang::format::(anonymous namespace)::Formatter::analyze(clang::format::TokenAnnotator&, llvm::SmallVectorImpl<clang::format::AnnotatedLine*>&, clang::format::FormatTokenLexer&) Format.cpp:0:0
 #8 0x0000005576626784 clang::format::TokenAnalyzer::process(bool) (/usr/local/bin/clangd+0x1f86784)
 #9 0x0000005576617790 std::_Function_handler<std::pair<clang::tooling::Replacements, unsigned int> (clang::format::Environment const&), clang::format::internal::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, unsigned int, unsigned int, unsigned int, llvm::StringRef, clang::format::FormattingAttemptStatus*)::$_10>::_M_invoke(std::_Any_data const&, clang::format::Environment const&) Format.cpp:0:0
#10 0x00000055765ff590 clang::format::internal::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, unsigned int, unsigned int, unsigned int, llvm::StringRef, clang::format::FormattingAttemptStatus*) (/usr/local/bin/clangd+0x1f5f590)
#11 0x000000557660046c clang::format::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, llvm::StringRef, clang::format::FormattingAttemptStatus*) (/usr/local/bin/clangd+0x1f6046c)
#12 0x0000005575ae6288 void llvm::detail::UniqueFunctionBase<void>::CallImpl<clang::clangd::ClangdServer::formatFile(llvm::StringRef, std::optional<clang::clangd::Range>, llvm::unique_function<void (llvm::Expected<clang::tooling::Replacements>)>)::$_0>(void*) ClangdServer.cpp:0:0
#13 0x0000005575c88c44 void llvm::detail::UniqueFunctionBase<void>::CallImpl<clang::clangd::TUScheduler::runWithSemaphore(llvm::StringRef, llvm::StringRef, llvm::unique_function<void ()>, clang::clangd::Semaphore&)::$_0>(void*) TUScheduler.cpp:0:0
#14 0x0000005575e34164 void* llvm::thread::ThreadProxy<std::tuple<clang::clangd::AsyncTaskRunner::runAsync(llvm::Twine const&, llvm::unique_function<void ()>)::$_1>>(void*) Threading.cpp:0:0
#15 0x0000007f8ca0ee58 (/lib/aarch64-linux-gnu/libc.so.6+0x7ee58)
#16 0x0000007f8ca77f9c (/lib/aarch64-linux-gnu/libc.so.6+0xe7f9c)
[Error - 9:08:26 PM] The Clang Language Server server crashed 1 times in the last 3 minutes. The server will not be restarted. See the output for more information.
[Error - 9:08:26 PM] Request textDocument/formatting failed.
[object Object]

```

> clang-format

```text
[user@host ~]% clang-format --dry-run libavutil/pixdesc.c
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.  Program arguments: clang-format --dry-run libavutil/pixdesc.c
 #0 0x000000555c68d56c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/llvm/bin/clang-format+0x7d56c)
 #1 0x000000555c68b718 llvm::sys::RunSignalHandlers() (/opt/llvm/bin/clang-format+0x7b718)
 #2 0x000000555c68dfbc SignalHandler(int) Signals.cpp:0:0
 #3 0x0000007fa4f357d0 (linux-vdso.so.1+0x7d0)
 #4 0x000000555c721f48 clang::format::WhitespaceManager::getCells(unsigned int, unsigned int) (/opt/llvm/bin/clang-format+0x111f48)
 #5 0x000000555c7214f4 clang::format::WhitespaceManager::alignArrayInitializers(unsigned int, unsigned int) (/opt/llvm/bin/clang-format+0x1114f4)
 #6 0x000000555c71e284 clang::format::WhitespaceManager::generateReplacements() (/opt/llvm/bin/clang-format+0x10e284)
 #7 0x000000555c6cbf40 clang::format::(anonymous namespace)::Formatter::analyze(clang::format::TokenAnnotator&, llvm::SmallVectorImpl<clang::format::AnnotatedLine*>&, clang::format::FormatTokenLexer&) Format.cpp:0:0
 #8 0x000000555c6edfe8 clang::format::TokenAnalyzer::process(bool) (/opt/llvm/bin/clang-format+0xddfe8)
 #9 0x000000555c6de704 std::_Function_handler<std::pair<clang::tooling::Replacements, unsigned int> (clang::format::Environment const&), clang::format::internal::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, unsigned int, unsigned int, unsigned int, llvm::StringRef, clang::format::FormattingAttemptStatus*)::$_10>::_M_invoke(std::_Any_data const&, clang::format::Environment const&) Format.cpp:0:0
#10 0x000000555c6c5a78 clang::format::internal::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, unsigned int, unsigned int, unsigned int, llvm::StringRef, clang::format::FormattingAttemptStatus*) (/opt/llvm/bin/clang-format+0xb5a78)
#11 0x000000555c6c6900 clang::format::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, llvm::StringRef, clang::format::FormattingAttemptStatus*) (/opt/llvm/bin/clang-format+0xb6900)
#12 0x000000555c6499e4 clang::format::format(llvm::StringRef) ClangFormat.cpp:0:0
#13 0x000000555c6478ec main (/opt/llvm/bin/clang-format+0x378ec)
#14 0x0000007fa4957780 (/lib/aarch64-linux-gnu/libc.so.6+0x27780)
#15 0x0000007fa4957858 __libc_start_main (/lib/aarch64-linux-gnu/libc.so.6+0x27858)
#16 0x000000555c644bf8 _start (/opt/llvm/bin/clang-format+0x34bf8)
Segmentation fault
```

**Comment by EugeneZelenko** [comment]
@valord577: Could you please provide reproducer?

**Comment by valord577** [comment]
@EugeneZelenko 

#### Steps to Reproduction:

1. File: [llvm-issue-86281.zip](https://github.com/llvm/llvm-project/files/14724715/llvm-issue-86281.zip)
2. `unzip llvm-issue-86281.zip && cd llvm-issue-86281`
3. (amd64 only) `docker run --privileged --rm tonistiigi/binfmt --install arm64`
4. `docker build --platform=linux/arm64 -t "llvm-issue:86281" .`
5. `docker run --rm -it --platform=linux/arm64 llvm-issue:86281`


```text
[user@host ~]% docker run --rm -it --platform=linux/arm64 llvm-issue:86281
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.  Program arguments: clang-format --dry-run pixdesc.c
 #0 0x00000055624fd56c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/llvm/bin/clang-format+0x7d56c)
 #1 0x00000055624fb718 llvm::sys::RunSignalHandlers() (/opt/llvm/bin/clang-format+0x7b718)
 #2 0x00000055624fdfbc SignalHandler(int) Signals.cpp:0:0
 #3 0x0000007fb547c7d0 (linux-vdso.so.1+0x7d0)
 #4 0x0000005562591f48 clang::format::WhitespaceManager::getCells(unsigned int, unsigned int) (/opt/llvm/bin/clang-format+0x111f48)
 #5 0x00000055625914f4 clang::format::WhitespaceManager::alignArrayInitializers(unsigned int, unsigned int) (/opt/llvm/bin/clang-format+0x1114f4)
 #6 0x000000556258e284 clang::format::WhitespaceManager::generateReplacements() (/opt/llvm/bin/clang-format+0x10e284)
 #7 0x000000556253bf40 clang::format::(anonymous namespace)::Formatter::analyze(clang::format::TokenAnnotator&, llvm::SmallVectorImpl<clang::format::AnnotatedLine*>&, clang::format::FormatTokenLexer&) Format.cpp:0:0
 #8 0x000000556255dfe8 clang::format::TokenAnalyzer::process(bool) (/opt/llvm/bin/clang-format+0xddfe8)
 #9 0x000000556254e704 std::_Function_handler<std::pair<clang::tooling::Replacements, unsigned int> (clang::format::Environment const&), clang::format::internal::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, unsigned int, unsigned int, unsigned int, llvm::StringRef, clang::format::FormattingAttemptStatus*)::$_10>::_M_invoke(std::_Any_data const&, clang::format::Environment const&) Format.cpp:0:0
#10 0x0000005562535a78 clang::format::internal::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, unsigned int, unsigned int, unsigned int, llvm::StringRef, clang::format::FormattingAttemptStatus*) (/opt/llvm/bin/clang-format+0xb5a78)
#11 0x0000005562536900 clang::format::reformat(clang::format::FormatStyle const&, llvm::StringRef, llvm::ArrayRef<clang::tooling::Range>, llvm::StringRef, clang::format::FormattingAttemptStatus*) (/opt/llvm/bin/clang-format+0xb6900)
#12 0x00000055624b99e4 clang::format::format(llvm::StringRef) ClangFormat.cpp:0:0
#13 0x00000055624b78ec main (/opt/llvm/bin/clang-format+0x378ec)
#14 0x0000007fb4e97780 (/lib/aarch64-linux-gnu/libc.so.6+0x27780)
#15 0x0000007fb4e97858 __libc_start_main (/lib/aarch64-linux-gnu/libc.so.6+0x27858)
#16 0x00000055624b4bf8 _start (/opt/llvm/bin/clang-format+0x34bf8)
```

**Comment by valord577** [comment]
@EugeneZelenko Hello, any progress please?

**Comment by owenca** [comment]
Fixed in cceedc939a43.
